packages/internal/src/types.d.ts

```diff
@@ -18,3 +20,17 @@ export declare class SyncCallback<
 
   public isSync: true;
 }
+
+declare const type: unique symbol;
```

Comment:
It's a little weird to me to use the identifier `type` here when it's also a TypeScript keyword.  Could you please name it something like `TypedMatcherParameter` or anything that you think would be better.?